conanfile.py

```diff
@@ -1,76 +1,128 @@
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get, rmdir
+from conan.tools.microsoft import is_msvc
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.53.0"
 
 
 class PROPOSALConan(ConanFile):
     name = "proposal"
     homepage = "https://github.com/tudo-astroparticlephysics/PROPOSAL"
-    license = "GNU Lesser General Public License v3.0"
-    description = "the very best lepton and photon propagator"
+    license = "LGPL-3.0"
+    package_type = "library"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "monte Carlo based lepton and photon propagator"
     topics = ("propagator", "lepton", "photon", "stochastic")
+
     settings = "os", "compiler", "build_type", "arch"
-    exports_sources = "*"
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
-        "with_testing": [True, False],
         "with_python": [True, False],
+        "with_testing": [True, False],
         "with_documentation": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
+        "with_python" : False,
         "with_testing": False,
-        "with_python": False,
         "with_documentation": False,
     }
-    generators = "cmake_find_package", "cmake_paths"
-    _cmake = None
+
+    @property
+    def _min_cppstd(self):
+        return "14"
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "msvc": "191",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5",
+        }
 
     def config_options(self):
         if self.settings.os == "Windows":
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self)
+        self.folders.generators = "build"
 
     def requirements(self):
-        self.requires("cubicinterpolation/0.1.5")
-        self.requires("spdlog/1.10.0")
-        self.requires("nlohmann_json/3.9.1")
+        # cubicinterpolation: headers are transitively included, and function calls are made
+        # from implementation in headers (templates)
+        self.requires("cubicinterpolation/0.1.5", transitive_headers=True, transitive_libs=True)
+        # spdlog: requires transitive_libs due to direct calls to functionality from headers
+        self.requires("spdlog/1.11.0", transitive_headers=True, transitive_libs=True)
+        # nlohmann_json: public headers include json.hpp and json_fwd.hpp
+        self.requires("nlohmann_json/3.11.2", transitive_headers=True)
         if self.options.with_python:
             self.requires("pybind11/2.10.1")
         if self.options.with_testing:
-            self.requires("boost/1.78.0")
+            self.requires("boost/1.75.0")
```

Comment:
why the downgrading from 1.78 to 1.75

Reply:
`cubicinterpolation` itself has a requirement of 1.75.0, and somehow this creates a conflict in conan when proposal defined a different version.
However, this is only relevant if one wants to build the tests. It shouldn't make any practical difference.